test/common/spin_barrier.h

```diff
@@ -52,13 +52,12 @@ void SpinWaitWhile(Predicate pred) {
             }
         }
     }
-    std::atomic_thread_fence(std::memory_order_acquire);
 }
 
 //! Spin WHILE the condition is true.
 template <typename T, typename C>
 void SpinWaitWhileCondition(const std::atomic<T>& location, C comp) {
-    SpinWaitWhile([&] { return comp(location.load(std::memory_order_relaxed)); });
```

Comment:
In termx of C++11 memory model, it is enough. However, standalone barriers are complex for tools (see https://github.com/google/sanitizers/issues/1415) and understanding (because they affect every previous atomic load and so on). Moreover, they do not bring anything in many of cases, especially, on x86.

Reply:
Sure. Thanks!